src/sink.rs

```diff
@@ -159,6 +197,40 @@ impl Sink {
         self.controls.pause.store(false, Ordering::SeqCst);
     }
 
+    // There is no `can_seek()` method as its impossible to use correctly. Between
+    // checking if a source supports seeking and actually seeking the sink can
+    // switch to a new source.
+
+    /// Attempts to seek to a given position in the current source.
+    ///
+    /// This blocks between 0 and ~5 milliseconds.
```

Comment:
Out of curiosity, what is the reason the 5ms delay? Is it related to the underlying file format?